pkg/apis/kubeflow.org/v2alpha1/trainjob_types.go

```diff
@@ -261,8 +265,30 @@ type TrainJobStatus struct {
 	// Conditions for the TrainJob.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// ReplicatedJobsStatus tracks the number of Jobs for each replicatedJob in TrainJob.
-	ReplicatedJobsStatus []jobsetv1alpha2.ReplicatedJobStatus `json:"replicatedJobsStatus,omitempty"`
+	// JobsStatus tracks the child Jobs in TrainJob.
+	JobsStatus []JobStatus `json:"jobsStatus,omitempty"`
+}
+
+type JobStatus struct {
```

Comment:
I think that there are not such a concern because this status will not sync with JobSet ones.
The runtimes could combined with except of JobSet like batch/v1 Job, and SlurmNode, or FluxMiniCluster something.

Reply:
How we can make sure that these status parameters (Ready, Succeeded, etc.) are applicable for other objects like SlurmNode, FluxMiniCluster?

Reply:
cc @vsoch

Reply:
> How we can make sure that these status parameters (Ready, Succeeded, etc.) are applicable for other objects like SlurmNode, FluxMiniCluster?

I will extend the runtime interface so that we can return the Jobs status. But the mechanism will be implemented after https://github.com/kubeflow/training-operator/pull/2298 and https://github.com/kubeflow/training-operator/pull/2295.

Reply:
I don't think the status parameters would be directly relevant for custom resources managed by another operator, but some higher level calling entity orchestrating something from JobSet/TrainJob and Flux MiniCluster could have awareness of the statuses/conditions. I would not scope Slurm or Flux to just have Training abstractions - the work can be more general than that.